### PR TITLE
Expose DWN PermissionsGrant Message

### PR DESCRIPTION
**Pull request by diehuxx**
Minimum permissions interface to expose to unblock delegated permissions grants.

`dwn.permissions.send()` will be extended in a future PR to to send other `Permissions` interface messages.

### CODE REVIEW

**Comment by frankhinek** [comment]
@diehuxx Reviewed the PR and had one question that I was hoping to get your insight on:

The approach to sending a `PermissionsGrant` is different than how both Record and ProtocolsConfigure messages are sent to a remote DWN.  What was the motivation for introducing a new `message` property to the send request object?

File: packages/api/src/dwn-api.ts [hunk 1]
```diff
@@ -91,6 +106,11 @@ export type RecordsWriteRequest = {
   store?: boolean;
 }
 
+export type PermissionGrantRequest = {
```
Comment (frankhinek) [inline packages/api/src/dwn-api.ts:109]: @diehuxx This type appears to be a duplicate.  It isn't used anywhere as best I can tell.  Should it be removed?

File: packages/api/src/dwn-api.ts [hunk 2]
```diff
@@ -19,6 +20,20 @@ import { DwnInterfaceName, DwnMethodName } from '@tbd54566975/dwn-sdk-js';
 import { Record } from './record.js';
 import { Protocol } from './protocol.js';
 import { dataToBlob } from './utils.js';
+import { PermissionsGrant } from '@tbd54566975/dwn-sdk-js';
+import { PermissionsGrantMessage } from '@tbd54566975/dwn-sdk-js';
+import { PermissionsGrantOptions } from '@tbd54566975/dwn-sdk-js';
```
Comment (frankhinek) [inline packages/api/src/dwn-api.ts:25]: Can these be consolidated into the existing import statement a few lines up?

File: packages/agent/src/dwn-manager.ts [hunk 3]
```diff
@@ -184,8 +188,11 @@ export class DwnManager {
       });
       dwnRpcRequest.message = message;
       messageData = data;
-
+    } else if ('message' in request) {
```
Comment (frankhinek) [inline packages/agent/src/dwn-manager.ts:191]: @diehuxx The approach to sending a `PermissionsGrant` is different than how both Record and ProtocolsConfigure messages are sent to a remote DWN. What was the motivation for introducing a new message property to the send request object?

File: packages/api/tests/dwn-api.spec.ts [hunk 4]
```diff
@@ -569,4 +571,57 @@ describe('DwnApi', () => {
       });
     });
   });
+
+  describe('permissions.grant()', () => {
```
Comment (frankhinek) [inline packages/api/tests/dwn-api.spec.ts:575]: Could a test for sending a `PermissionsGrant` to a remote DWN be added in this PR?

File: packages/agent/src/dwn-manager.ts (hunk 3 above)
Comment (diehuxx) [reply packages/agent/src/dwn-manager.ts:191]: I've been wanting to make that addition separate of `PermissionsGrants`. The existing code forces an existing message to be turned into `options` then re built and signed. Why not just send the existing message?

File: packages/api/tests/dwn-api.spec.ts (hunk 4 above)
Comment (diehuxx) [reply packages/api/tests/dwn-api.spec.ts:575]: If I understand agents correctly, setting `dwn.permissions.grant(target  : bobDid.did, ...)` will send to a remote. We do that in the test in `describe('target: did'`

**Pull request closed without merging**